### DISCUSSION TITLE
Where is model 'en_core_web_sm'?

### DISCUSSION THREAD

**Discussion by mcswell** [explosion/spaCy#10959] [Help: Other Questions] [answered]
We have Python 3.6 and spacy 3.3.1 installed on a server.  After 
    `import spacy`
we have a line
    `spacy.load("en_core_web_sm")`
For some of us, this works, but others get the familiar and dreaded msg
    `Can't find model 'en_core_web_sm'.  It doesn't seem to be a Python
     package or a valid path to a data directory.`

Why would this identical code work for some of us but not for others?  My initial guess was that it had to do with the PYTHONPATH environment variable (we're running this under Linux), but in fact we all have the same path: 
    `:/usr/local/lib64/python3.6//:/usr/local/lib/python3.6//:`
So that can't be the problem.

Where exactly is this model stored?  There doesn't seem to be any file en_core_web_sm.py (or .pyc) (which agrees with the error msg about it not being a Python package), so it's hard to see what the path *should* be.  I tried tracing the code back in 
    `usr/local/lib64/python3.6/site-packages/spacy/util.py`
but it quickly goes to the generic importlib.import_module(name) function, with no indication of how it expects to convert the supplied string 'en_core_web_sm' into an actual file.

A more complete error msg would be something like "looked for file XXX in /usr/local/lib64/python... and in /user/local/lib/python..., but couldn't find this model".

**Comment by mcswell** [comment, marked as answer]
Solved my own problem (:).  When I first installed this model, I did
    `python3 -m spacy download en_core_web_sm`
which installed it locally, for me.  I should have prefaced it with `sudo`, then it would have installed into the right place.

I figured this out by looking back through my history to see the full cmd I used to load it.  I spent awhile longer trying to figure out how I would have known without looking at my history.  Unlike a library module, the string repr of the model is not particularly helpful:
    `<spacy.lang.en.English object at 0x7fa9d007ad68>`
In particular, it doesn't say where it loaded from (unlike modules, which show the path they loaded from).  It took me awhile, but eventually:
    `f = spacy.load('en_core_web_sm')
    f._path`
shows the path.  I would then have known that it was coming from under my home dir...

**Reply by polm** [reply]
Glad you figured it out!
